fish/Containerfile

```diff
@@ -0,0 +1,13 @@
+# Get RHCOS base image of target cluster `oc adm release info --image-for rhel-coreos`
+# This example uses rhel-coreos image from 4.13.0-rc.0
+FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:42e45dd0ba1be3d2681972d6d38c42008c70ddb8c5a96f1f770a11f9bbfb048f
+
+# RHEL entitled host is needed here to access RHEL packages
+# Create symlink to have RHEL entitlement and redhat.repo available at correct location inside container build
+RUN ln -s /run/secrets/redhat.repo /etc/yum.repos.d/redhat.repo && \
```

Comment:
This line shouldn't be needed, as the /run/secrets/redhat.repo file is not used. The /etc/yum.repos.d/redhat.repo file is generated on the fly. Here's an example with regular ubi9

```
# podman run --rm -ti ubi9
[root@228daa45e777 /]# head -n2 /etc/yum.repos.d/redhat.repo
head: cannot open '/etc/yum.repos.d/redhat.repo' for reading: No such file or directory

[root@228daa45e777 /]# yum
Updating Subscription Management repositories.
Unable to read consumer identity
Subscription Manager is operating in container mode.
usage: yum [options] COMMAND
[...]

[root@228daa45e777 /]# tail -n2 /etc/yum.repos.d/redhat.repo
metadata_expire = 86400
enabled_metadata = 0

[root@228daa45e777 /]# echo "# test" > /etc/yum.repos.d/redhat.repo && tail -n2 /etc/yum.repos.d/redhat.repo
# test


[root@228daa45e777 /]# yum
Updating Subscription Management repositories.
[...]

[root@228daa45e777 /]# tail -n2 /etc/yum.repos.d/redhat.repo
metadata_expire = 86400
enabled_metadata = 0
```

However, your example does need this line to work with 4.13.0-rc.0, because the RHCOS development build is using CentOS Stream content. This seems to cause RHCOS to identify itself as CentOS and not RHEL, so redhat.repo is not generated. Once RHCOS is using RHEL content again this should not be needed (hopefully).

Reply:
Great feedback, not using symlink would make things more sane. I will recheck once we have rhel9 based content available and make adjustment if it is possible.